Eventnet/Eventnet.Api/Controllers/UserAccountController.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Eventnet.DataAccess;
+using Eventnet.Domain;
+using Eventnet.Models;
+using Eventnet.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Eventnet.Controllers;
+
+[Route("api/auth")]
+public class UserAccountController : Controller
+{
+    private readonly UserManager<ApplicationUser> userManager;
+    private readonly RoleManager<IdentityRole> roleManager;
+    private readonly IJwtAuthService jwtAuthService;
+    private readonly IEmailService emailService;
+
+    public UserAccountController(UserManager<ApplicationUser> userManager,
+        RoleManager<IdentityRole> roleManager,
+        IJwtAuthService jwtAuthService,
+        IEmailService emailService)
+    {
+        this.userManager = userManager;
+        this.roleManager = roleManager;
+        this.jwtAuthService = jwtAuthService;
+        this.emailService = emailService;
+    }
+
+    [HttpPost("login")]
+    [Produces(typeof(LoginResult))]
+    public async Task<IActionResult> Login([FromBody] LoginModel loginModel)
+    {
+        if (!ModelState.IsValid)
+            return UnprocessableEntity(ModelState);
+
+        var user = await userManager.FindByNameAsync(loginModel.UserName)
+            ?? await userManager.FindByEmailAsync(loginModel.Email);
+
+        if (user == null || !await userManager.CheckPasswordAsync(user, loginModel.Password))
```

Comment:
Возможно имеет смысл разбить на два if.

Reply:
Нет особой нужды, так как ответ всегда будет одним и тем же, потому что мы не хотим сообщать пользователю, что пароль неверен

Reply:
Я из соображений красоты когда говорил. Имелось в виду следующее:
```suggestion
        if (user == null)
            return Unauthorized();
        var isCorrectPassword = await userManager.CheckPasswordAsync(user, loginModel.Password);
        if (!isCorrectPassword)
            return Unauthorized();
        
```